libdnf/advisory/advisory_query.cpp

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/advisory/advisory_query.hpp"
+
+#include "libdnf/advisory/advisory_package_private.hpp"
+#include "libdnf/rpm/package_set.hpp"
+#include "libdnf/rpm/solv/package_private.hpp"
+#include "libdnf/rpm/solv_sack_impl.hpp"
+
+#include <solv/evr.h>
+
+// For glob support
+#include <fnmatch.h>
+
+namespace libdnf::advisory {
+
+AdvisoryQuery::AdvisoryQuery(AdvisorySack * sack) : advisory_sack(sack), data_map(sack->data_map){};
+
+AdvisoryQuery::~AdvisoryQuery() = default;
+
+AdvisoryQuery & AdvisoryQuery::ifilter_name(const std::string & pattern, sack::QueryCmp cmp_type) {
+    const std::vector<std::string> patterns = {pattern};
+    ifilter_name(patterns, cmp_type);
+    return *this;
+}
+
+AdvisoryQuery & AdvisoryQuery::ifilter_name(const std::vector<std::string> & patterns, sack::QueryCmp cmp_type) {
```

Comment:
I think it should primarily be consistent with other queries (`SolvQuery` and `RepoQuery`) which it is now, they all filter out everything. 

Other than that I would be fine either way. Returning empty does make sense to me, since we are matching the patterns and when there are no patters -> nothing matches. On the other hand return everything might be a bit more practical sometimes.

Reply:
I also don't know what would be the best behavior even for rpm queries. There is also an option to throw an exception when the empty vector is provided.